### PR TITLE
MQE: Add extra comparison op edge case tests

### PR DESCRIPTION
**Pull request by jhesketh**
Related to https://github.com/grafana/mimir/pull/9398

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [x] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Comment by jhesketh** [comment]
(I'll update CHANGELOG after https://github.com/grafana/mimir/pull/9417 merges)

File: pkg/streamingpromql/engine_test.go [hunk 1]
```diff
@@ -1889,3 +1889,30 @@ func TestCompareVariousMixedMetricsRate(t *testing.T) {
...
+	// Test each label individually to catch edge cases in with single series
+	labelCombinations := testutils.Combinations(labelsToUse, 1)
+	// Generate combinations of 2 labels. (e.g., "a,b", "e,f", etc)
+	labelCombinations = append(labelCombinations, testutils.Combinations(labelsToUse, 2)...)
+
+	expressions := []string{}
+
+	for _, labels := range labelCombinations {
+		labelRegex := strings.Join(labels, "|")
+		for _, op := range []string{"==", "!=", ">", "<", ">=", "<="} {
```
Comment (charleskorn) [inline pkg/streamingpromql/engine_test.go:1905]: This will only test vector / scalar combinations. Should we also add some cases for vector / vector combinations?

Comment (jhesketh) [reply pkg/streamingpromql/engine_test.go:1905]: Good idea, done.

Are we testing scalar / scalar cases anywhere? (they should't go here, but I only recall a few basic comparisons upstream)

Comment (charleskorn) [reply pkg/streamingpromql/engine_test.go:1905]: >Are we testing scalar / scalar cases anywhere? (they should't go here, but I only recall a few basic comparisons upstream)

Yep, [here](https://github.com/grafana/mimir/blob/45a14c913b0433003b28d17cc1f958dcfe72fd14/pkg/streamingpromql/testdata/ours/binary_operators.test#L703-L720).

**Review by charleskorn — APPROVED** [review]
